### ISSUE TITLE
Simplify UpgradableConstructor use of native lifecycle callbacks

### ISSUE THREAD

**Issue by nolanlawson** [salesforce/lwc#3818]
(This bug only applies to native custom element lifecycle mode.)

Currently the UpgradableConstructor uses a WeakMap to keep track of lifecycle callbacks for FACE:

https://github.com/salesforce/lwc/blob/187a24dd43ae74c58cea8eb404e67cf7c6c78046/packages/%40lwc/engine-dom/src/custom-elements/create-custom-element.ts#L45-L48

However, it doesn't do this for connected/disconnected callbacks:

https://github.com/salesforce/lwc/blob/187a24dd43ae74c58cea8eb404e67cf7c6c78046/packages/%40lwc/engine-dom/src/custom-elements/create-custom-element.ts#L97-L111

This is technically a bug, since the connected/disconnected callbacks effectively are created once per constructor and never change per instance. So you might think that the connected/disconnected callbacks should use the same `WeakMap` system as the FACE callbacks, in order to fix the bug.

However, it doesn't actually matter in practice, because every callback closure just passes in `this`/`elm`, which is unique per instance:

https://github.com/salesforce/lwc/blob/187a24dd43ae74c58cea8eb404e67cf7c6c78046/packages/%40lwc/engine-core/src/framework/rendering.ts#L340-L357

(To get the `vm`, the underlying callbacks call `getAssociatedVM(elm)`.)

Here is a test that confirms that the current behavior is correct: https://github.com/salesforce/lwc/commit/687572eb7c821c13ee156accade8b9f18cc97654

So basically, the `WeakMap`s are useless at best and a perf hit at worst. It should be possible to greatly simplify this code.
